VoxelSniperCore/src/main/java/com/github/kevindagame/brush/BrushBuilder.kt

```diff
@@ -0,0 +1,57 @@
+package com.github.kevindagame.brush
+
+import java.util.*
+import java.util.function.Supplier
+
+/**
+ * Builder class for (old) brushes
+ */
+class BrushBuilder {
+    private lateinit var name: String
+    private lateinit var permission: String
+    private val aliases: MutableList<String> = mutableListOf()
+    private lateinit var supplier: Supplier<out IBrush>
+
+    /**
+     * Sets the name of the brush.
+     */
+    fun name(name: String): BrushBuilder
+    {
+        this.name = name
+        return this
+    }
+
+    /**
+     * Add aliases for the brush.
+     */
+    fun alias(vararg aliases: String): BrushBuilder
+    {
+        this.aliases.addAll(aliases)
+        return this
+    }
+
+    /**
+     * Sets the permission of the brush.
+     */
+    fun setPermission(permission: String): BrushBuilder
+    {
+        this.permission = permission
+        return this
+    }
+
+    /**
+     * Sets the class of the brush.
+     */
+    fun setSupplier(supplier: Supplier<out IBrush>): BrushBuilder
+    {
+        this.supplier = supplier
+        return this
+    }
+
+    fun build(): BrushData {
+        if(!::permission.isInitialized) {
+            permission = "voxelsniper.brush.${name.lowercase(Locale.getDefault())}"
+        }
+        return BrushData(name, permission, aliases, supplier)
+    }
```

Comment:
A builder provides more customizability and free order of parameters, especially as we add more brush types to polyBrush over time. And I dislike long constructors. (plus that a builder allows for default values such as the permission, which is logic that in my opinion should not be on the brush itself)

Reply:
Kotlin already checks for unset parameters. It will throw an exeption with a clear stacktrace, which is sufficient in my opinion.
![image](https://user-images.githubusercontent.com/65958288/225908482-bac462d1-1673-4723-8e65-66a123756d48.png)